internal/controller/dscinitialization/auth.go

```diff
@@ -2,110 +2,49 @@ package dscinitialization
+	"github.com/opendatahub-io/opendatahub-operator/v2/api/common"
 	serviceApi "github.com/opendatahub-io/opendatahub-operator/v2/api/services/v1alpha1"
-	"github.com/opendatahub-io/opendatahub-operator/v2/internal/controller/components/dashboard"
-	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster/gvk"
-	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/common"
+	"github.com/opendatahub-io/opendatahub-operator/v2/pkg/cluster"
 )
 
-func (r *DSCInitializationReconciler) createAuth(ctx context.Context, dscInit *dsciv1.DSCInitialization) error {
-	log := logf.FromContext(ctx)
-	// check for the dashboardConfig kind.
-	// Once the groupsConfig entry in the dashboardConfig is removed this logic can be removed.
-	crd := &apiextv1.CustomResourceDefinition{}
-	err := r.Client.Get(ctx, client.ObjectKey{Name: "odhdashboardconfigs.opendatahub.io"}, crd)
-
-	// can't find the dashboardConfig kind so create the default.
-	if err != nil && k8serr.IsNotFound(err) {
-		log.Info("couldn't find odhDashboardConfig CRD, creating auth with defaults.")
-		err = r.Client.Create(ctx, buildDefaultAuth())
-		if err != nil && !k8serr.IsAlreadyExists(err) {
-			return err
-		}
-		return nil
-	} else if err != nil {
-		return fmt.Errorf("failed to get odhdashboardconfigs.opendatahub.io CRD: %w", err)
-	}
-
-	// check if a dashboardConfig instance exists.
-	odhObject := &unstructured.Unstructured{}
-	odhObject.SetGroupVersionKind(gvk.OdhDashboardConfig)
-	err = r.Client.Get(ctx, client.ObjectKey{
-		Name:      "odh-dashboard-config",
-		Namespace: dscInit.Spec.ApplicationsNamespace,
-	}, odhObject)
-	if err != nil && k8serr.IsNotFound(err) {
-		log.Info("couldn't find odhDashboardConfig instance, creating auth with defaults.")
-		err = r.Client.Create(ctx, buildDefaultAuth())
-		if err != nil && !k8serr.IsAlreadyExists(err) {
-			return err
-		}
-		return nil
-	} else if err != nil {
-		return fmt.Errorf("failed to get odh-dashboard-config instance: %w", err)
+var (
+	adminGroups = map[common.Platform]string{
+		cluster.SelfManagedRhoai: "rhods-admins",
+		cluster.ManagedRhoai:     "dedicated-admins",
+		cluster.OpenDataHub:      "odh-admins",
 	}
+)
 
-	// dashboardConfig CRD exists and we have an instance so copy the groups to the auth CR
-	foundGroups, ok, err := unstructured.NestedStringMap(odhObject.Object, "spec", "groupsConfig")
-	if err != nil {
-		return err
-	}
-	if ok {
-		adminGroup := []string{}
-		allowedGroup := []string{}
-		added := common.AddMissing(&adminGroup, foundGroups["adminGroups"])
-		added += common.AddMissing(&allowedGroup, foundGroups["allowedGroups"])
+func (r *DSCInitializationReconciler) createAuth(ctx context.Context, platform common.Platform) error {
+	// if Auth CR exsits, we do nothing
+	a := serviceApi.Auth{}
 
-		// only update if we found a new group in the list
-		if added == 0 {
+	if err := r.Client.Get(ctx, client.ObjectKey{Name: serviceApi.AuthInstanceName}, &a); err != nil {
```

Comment:
is this get required ? the Create would fail if the object already exists

Reply:
the logic was wrong. if should be on `!k8serr.IsNotFound(err) {...} `
i did update.